ReactiveCocoa/Swift/Signal.swift

```diff
@@ -569,6 +569,72 @@ public func takeWhile<T, E>(predicate: T -> Bool)(signal: Signal<T, E>) -> Signa
 	}
 }
 
+/// Throttle values sent by the receiver, so that at least `interval`
+/// seconds pass between each, then forwards them on the given scheduler.
+///
+/// If multiple values are received before the interval has elapsed, the
+/// latest value is the one that will be passed on.
+public func throttle<T, E>(interval: NSTimeInterval, onScheduler scheduler: DateSchedulerType)(signal: Signal<T, E>) -> Signal<T, E> {
+	precondition(interval >= 0)
+
+	return Signal { observer in
+		let state = Atomic(ThrottleState<T>())
+		let flush: () -> Void = {
+			let (_, doSend: (() -> Void)?) = state.modify { (var state) in
```

Comment:
Could this maybe yield `T?` instead of the closure, to keep the signature simpler and easier to understand?